examples/d2n-extended/flake.nix

```diff
@@ -0,0 +1,38 @@
+{
+  inputs = {
+    dream2nix.url = "github:nix-community/dream2nix";
+    src.url = "github:BurntSushi/ripgrep/13.0.0";
+    src.flake = false;
+  };
+
+  outputs = {
+    self,
+    dream2nix,
+    src,
+  } @ inp: let
+    dream2nix = inp.dream2nix.lib2.init {
+      systems = ["x86_64-linux"];
+      config.projectRoot = ./.;
+      config.extra = {
+        subsystems.rust = {
+          builders.crane-new = "${inp.dream2nix}/src/subsystems/rust/builders/crane";
```

Comment:
While I understand this tests that the actual API doesn't explode, I think this still leaves the issue of this not actually testing whether the subsystem elements get overridden and more importantly – doesn't actually test that a new subsystem that doesn't come built-in will work.

We had this issue during the back and forth on this PR where this obscured the config serialisability issue – the test worked, but an external subsystem didn't work due to `extra`s not being present in the `config` when trying to run an impure translator.

As such, I feel we should actually have a test adding a new translator, that's not in the sources of dream2nix. It can be a dummy translator that's creating dummy packages saying `Hello World` or whatever, but at least should verify an external translator works end-to-end. I don't know it's worth blocking this PR over it or not, but at the very least it should be a short-term follow-up.

Reply:
Does the example I added seem fine to you?

Reply:
We discussed it on the matrix channel; it seems I was misremembering the exact issue, but if you say this tests what the actual issue was then and you have verified that this test fails if you break your code in the way it caused that issue – then I am fine with such an example.